vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridSingleSelectionIT.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid.it;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.vaadin.flow.component.grid.testbench.GridElement;
+import com.vaadin.flow.testutil.AbstractComponentIT;
+import com.vaadin.flow.testutil.TestPath;
+
+@TestPath("grid-single-selection")
+public class GridSingleSelectionIT extends AbstractComponentIT {
+
+    @Test
+    public void checkDeselectionAllowedByDefault() {
```

Comment:
"By default" here and in the comment below is incorrect. Because the `buildGrid` method calls `setDeselectAllowed` for both Grids, there is no test case ensuring how it works by default. Currently, these tests don't ensure that it works correctly without calling the setter.

I suggest not to call that setter in `buildGrid` for this first Grid.

Reply:
Fixed